recipes/qpdf/all/conanfile.py

```diff
@@ -0,0 +1,164 @@
+                raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support.")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/3.24.1")
+        self.tool_requires("pkgconf/1.9.3")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_STATIC_LIBS"] = not self.options.shared
+        # https://qpdf.readthedocs.io/en/latest/installation.html#build-time-crypto-selection
+        tc.variables["USE_IMPLICIT_CRYPTO"] = False
+        tc.cache_variables["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = True
+        if self.options.with_crypto == "native":
+            tc.variables["REQUIRE_CRYPTO_NATIVE"] = True
+            tc.variables["REQUIRE_CRYPTO_GNUTLS"] = False
+            tc.variables["REQUIRE_CRYPTO_OPENSSL"] = False
+        elif self.options.with_crypto == "openssl":
+            tc.variables["REQUIRE_CRYPTO_NATIVE"] = False
+            tc.variables["REQUIRE_CRYPTO_GNUTLS"] = False
+            tc.variables["REQUIRE_CRYPTO_OPENSSL"] = True
+        tc.generate()
+        tc = PkgConfigDeps(self)
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+        tc = VirtualBuildEnv(self)
+        tc.generate(scope="build")
+        # TODO: after https://github.com/conan-io/conan/issues/11962 is solved, remove the following workaround
+        env = Environment()
+        env.prepend_path("PKG_CONFIG_PATH", self.generators_folder)
+        envvars = env.vars(self)
+        envvars.save_script("pkg_config")
+
+    def _patch_sources(self):
+        # patch 0001 and 0002 are uniquely appliable, based ony dependency config
+        patches_to_apply = ["0003-exclude-unnecessary-cmake-subprojects.patch"]
+        if self.options.with_crypto == "openssl":
+            patches_to_apply.append("0002-libqpdf-cmake-deps-jpeg-zlib-openssl.patch")
+        else:
+            patches_to_apply.append("0001-libqpdf-cmake-deps-jpeg-zlib.patch")
+        for patch_file in patches_to_apply:
+            patch(self, base_path=self.source_folder,
+                  patch_file=os.path.join(self.source_folder, "..", "patches", patch_file))
+
+    def build(self):
+        self._patch_sources()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, pattern="LICENSE.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["qpdf"]
```

Comment:
Should this be `libqpdf`?

```suggestion
        self.cpp_info.libs = ["libqpdf"]
```

Reply:
as soon as i add this "lib" before it, it fails locally with:

```
CMake Error at build/generators/cmakedeps_macros.cmake:39 (message):
  Library 'libqpdf' not found in package.  If 'libqpdf' is a system library,
  declare it with 'cpp_info.system_libs' property
Call Stack (most recent call first):
  build/generators/qpdf-Target-release.cmake:34 (conan_package_library_targets)
  build/generators/qpdfTargets.cmake:26 (include)
  build/generators/qpdf-config.cmake:10 (include)
  CMakeLists.txt:5 (find_package)
```

I encountered previously that cmake does add in some scenarios the "lib"-keyword automatically. I don't know if this is here part of the magic that happens. Nevertheless it seems to correctly map with "qpdf" to libqpdf.a file on the disk.

Reply:
Nevermind, my bad. The install command changes the name to `qpdf`.

Reply:
Mh it seems you are right, that the package definition is not correctly, since some plattform tests still fails. 

When i reapply 
```
        self.cpp_info.components["libqpdf"].names["cmake_find_package"] = "libqpdf"
        self.cpp_info.components["libqpdf"].names["cmake_find_package_multi"] = "libqpdf"
```
and use for cpp_info.libs = "libqpdf", than the cmake found error dissolves, but the linker does not resolve correctly the symbols anymore again.

I am here quite out of knowledge how the CMakeLists configuration should be designed via conanfile, to match the official one. @jwillikers can you provide a tested setup for the package_info setup, that is in sync with the qpdf upstream cmake file? Otherwise i would like to just revert the the working state before that is not in sync with qpdf upstream, but was able to get the ci-green on all plattforms.

Reply:
Yep, the components have to define relationships to dependencies explicitly. Oh and, ` self.cpp_info.libs = ["qpdf"]` is correct. The `qpdf` project renames the library's output name to `qpdf` which I'd missed.